### PR TITLE
ERB the template.yaml

### PR DESCRIPTION
**Pull request by zachmargolis**
This is an attempt to make `template.yaml` easier to maintain (instead of relying on humans to copy-paste), would love feedback on it!

I chose the approach of: people making changes that affect it will need to remember to re-generate and check those changes in
- Pros: we get the full file in source and it's easy to inspect
- Cons: it's an extra step, and a surprise whiny build if you don't expect this

Alternative options: Update the CI scripts to gen this file on the fly, so we don't need to keep both the template and the source checked in

### CODE REVIEW

File: source/template.yaml [hunk 1]
```diff
@@ -54,17 +58,45 @@ Globals:
       environment: !Ref environment
 
 Resources:
+  # add all functions that need gitsha alias to this list
```
Comment (zachmargolis) [inline source/template.yaml:61]: this is a reordering (since there was a group of 3 functions together, and a fourth up top), not an actual change

**Comment by aduth** [comment]
I'd lean more to the side of not having this checked in since it's generated, easy to miss, and presumably not important for local development aside from the note of being able to inspect it easily? Could a compromise be to document how one could run the script themselves locally and inspect the file, even if it's ignored from git?

**Comment by zachmargolis** [comment]
> I'd lean more to the side of not having this checked in since it's generated, easy to miss, and presumably not important for local development aside from the note of being able to inspect it easily? Could a compromise be to document how one could run the script themselves locally and inspect the file, even if it's ignored from git?

I went ahead used this approach, and documented how to see the file in the README

File: source/template.yaml [hunk 2]
```diff
@@ -1,615 +0,0 @@
...
-    Value: !GetAtt ProofResolutionFunction.Arn
-  ProofResolutionFunctionIamRole:
-    Description: "Proof Resolution Role created for Proof Resolution function"
-    Value: !GetAtt ProofResolutionFunctionRole.Arn
-
-  ProofResolutionMockFunction:
-    Description: "Proof Resolution Mock Lambda Function ARN"
-    Value: !GetAtt ProofResolutionMockFunction.Arn
-  ProofResolutionMockFunctionIamRole:
-    Description: "Proof Resolution Mock Role created for Proof Address Mock function"
```
Comment (aduth) [inline source/template.yaml:614]: Observing that this changes to "...for Proof Resolution Mock function", which seems reasonable given the name of the function.

File: source/template.yaml [hunk 3]
```diff
@@ -1,615 +0,0 @@
...
-          - '{{resolve:ssm:/${Environment}/network/outboundproxy/no_proxy:1}}'
-          - Environment: !Ref environment
-        ENVIRONMENT_NAME: !Ref environment
-    Tags:
-      environment: !Ref environment
-
-Resources:
-  ProofAddressFunction:
-    Type: AWS::Serverless::Function 
-    Properties:
```
Comment (aduth) [inline source/template.yaml:59]: Observing that we're including `MemorySize: 128` and `VersionDescription: !Ref gitsha` properties in the new generated output that did not exist previously, which seems reasonable given the common configuration.

**Review by aduth — APPROVED** [review]
Question: Do we need to include commands to generate this file in `source/buildspec.yml`? I notice a few references to the `template.yaml` file there.

**Comment by zachmargolis** [comment]
> Question: Do we need to include commands to generate this file in `source/buildspec.yml`? I notice a few references to the `template.yaml` file there.

In CircleCI I added steps to build that file before the buildspec gets run

**Comment by crissupb** [comment]
No need to add it to the buildspec.yml.  The template gets copied up to S3 as part of the build.

**Review by crissupb — APPROVED** [review]
Looks greate.

**Pull request merged**